brushfire-training/src/main/scala/com/stripe/brushfire/local/CrossEntropyExample.scala

```diff
@@ -0,0 +1,44 @@
+package com.stripe.brushfire.local
+
+import com.stripe.bonsai._
+import com.stripe.brushfire._
+import com.twitter.algebird._
+
+import AnnotatedTree.{AnnotatedTreeTraversal, fullBinaryTreeOpsForAnnotatedTree}
+
+object CrossEntropyExample extends Defaults {
+
+  def main(args: Array[String]) {
+    val path = args.head
+    val cols = args.tail.toList
+
+    // T = Map[String, Long]
+    // P = String
+    // E = Double
+
+    val trainingData = Lines(path).map { line =>
+      val parts = line.split(",").reverse.toList
+      val label = parts.head
+      val values = parts.tail.map { s => s.toDouble }
+      Instance(line, 0L, Map(cols.zip(values): _*), Map(label -> 1L))
+    }.toIterable
+
+    val totalData: Map[String, Long] = Monoid.sum(trainingData.iterator.map(_.target))
```

Comment:
yeah, that's fine. We just need an estimate of the "true" entropy. We should use the exact same set that we are measuring the error on below.

I guess we could fit it into the semigroup approach by aggregating the total label distribution as we go. (in that way, by the way), it seems like `Error` should really be a semigroup and a function (maybe a full aggregator).